task_1.py

```diff
@@ -0,0 +1,40 @@
+class Product:
+    """Product class implementation"""
+
+    def __init__(self, name: str, price: float) -> None:
+        """Initialize a product instance"""
+
+        self.name = name
+        self.price = price
+
+    def get_totalprice(self, quantity: float) -> float:
+        """Return total price for a specified quantity of goods"""
+
+        return round(self.price * quantity, 2)
+
+
+class ShoppingCart():
+    """ShoppingCart class implementation"""
+
+    def __init__(self) -> None:
+        """Initialize a shopping cart instance"""
+
+        self.goods = []
+        self.buyed_quantity = []
+
+    def add_cartitem(self, product: Product, quantity: float) -> list:
+        """Combine products instances and corresponding purchased quantities"""
+
+        if product not in self.goods:
+            self.goods.append(product)
+            self.buyed_quantity.append(quantity)
+        else:
+            self.buyed_quantity[self.goods.index(product)] = self.buyed_quantity[self.goods.index(product)] + quantity
```

Comment:
Тут усе норм, але не пиши так - бо тебе колеги будуть ненавидиті за таке.
```suggestion
            idx = self.goods.index(product)
            self.buyed_quantity[idx] += quantity
```
Погодься що так це набагато легше прочитати

Reply:
@shorodilov так, я згоден. Код читабельніший. Я свідомо так зробив, щоб не нагромаджувати код змінними, без яких можна обійтися. Створення змінних не впливає на продуктивність?

Reply:
Ні, не впливає. Зміна то є лише поіменований вказівник на область пам'яті для подальшого звернення.
Самі по собі значення тут всеодно будуть створені та занесені до пам'яті